README.md

```diff
@@ -41,6 +43,61 @@ fn main() {
 }
 ```
 
+### Generate a new GPX file
+This example only generates tracks. You can add waypoints and routes as well by instantiating new ``Waypoint``s and ``Route``s.
+```rust
+use std::path::Path;
+use gpx::{Gpx, Track, TrackSegment, Waypoint, Route};
+use geo_types::{Point, coord};
+
+pub fn to_gpx<P: AsRef<Path>>(out_path: P) -> Result<String, Box<dyn Error>> {
+    // Instantiate Gpx struct
+    let mut gpx: Gpx = Gpx {
```

Comment:
Let's build this in the other order (`let track_segment = ...; let track = ...; let gpx = ...`). That way wou don't need indexing and can do directly e.g. `vec![track]`.